pkg/addonfactory/addonfactory.go

```diff
@@ -34,9 +34,11 @@ type AgentAddonFactory struct {
 	getValuesFuncs    []GetValuesFunc
 	agentAddonOptions agent.AgentAddonOptions
 	// trimCRDDescription flag is used to trim the description of CRDs in manifestWork. disabled by default.
-	trimCRDDescription    bool
-	hostingCluster        *clusterv1.ManagedCluster
-	agentInstallNamespace func(addon *addonapiv1alpha1.ManagedClusterAddOn) string
+	trimCRDDescription          bool
+	hostingCluster              *clusterv1.ManagedCluster
+	agentInstallNamespace       func(addon *addonapiv1alpha1.ManagedClusterAddOn) string
+	createAgentInstallNamespace bool
```

Comment:
If set this true, will the ns be deleted when disabling the addon?

Reply:
if the addon developers put the agent namespace yaml file into the manifests, the ns also can be managed by manifestwork. so could you give more explanation as to why this createAgentInstallNamespace is necessary?